src/transaction/buffer.rs

```diff
@@ -274,6 +298,10 @@ enum BufferEntry {
     Put(Value),
     // Value has been deleted.
     Del,
+    // Key should be check not exists before.
+    Insert(Value),
+    // Key should be check not exists before.
+    CheckNotExist,
```

Comment:
I think we can also provide a `check_not_exist` method.

Reply:
It's kind of unclear for me to see any possible use cases of a `check_not_exist()` method on a `Transaction`. So, for now, I use `CheckNotExist` mutation to handle "delete-your-writes" cases as [it was introduced](https://github.com/pingcap/tidb/pull/14968).
Do you have any idea about a possible use case of `check_not_exist` method?

Reply:
I'm not sure. @andylokandy Have you seen such requirements? IIRC there are people who want this.

Reply:
It's actually an optimistic version of key_exists() right? I think it's rarely that such optimization is need to expose to api level.